FactoryWeather/App/VIP Search/DataSource/SearchDataSource.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  SearchDataSource.swift
+//  FactoryWeather
+//
+//  Created by Matej Korman on 14/01/2019.
+//  Copyright © 2019 Matej Korman. All rights reserved.
+//
+
+import Foundation
+
+class SearchDataSource: DataSourceProtocol {
+    var sections = [SearchSection]()
+    private var locations = [Location]()
+    private var filterText: String?
+    
+    func addLocations(_ locations: [Location]) {
```

Comment:
As a convention, I would always extract `public`/`internal` methods or stored/computed properties into a separate extension of the host type.

In example, `addLocations` and `filterLocations` would be extracted into a `public` or `internal`(depending on the requirement) extension.

As a result, we get increased readability and understanding of the type's `API`.

Reply:
Can I write just `extension SearchDataSource` or should it be `internal extension SearchDataSource`?